src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -557,13 +562,14 @@ const WorkflowView = _.flow(
       [type === processAllAsSet, () => `1 ${rootEntityType} containing all ${entityMetadata[baseEntityType]?.count || 0} ${baseEntityType}s ${newSetMessage}`],
       [type === chooseSets, () => !!count ?
         `${count} selected ${rootEntityType}s ${newSetMessage}` :
-        `No ${rootEntityType}s selected`]
+        `No ${rootEntityType}s selected`],
+      [type === processSnapshotTable, () => `process entire snapshot table`]
     )
   }
 
   canSave() {
-    const { modifiedConfig: { rootEntityType } } = this.state
-    return this.isSingle() || !!rootEntityType
+    const { modifiedConfig: { rootEntityType, dataReferenceName } } = this.state
+    return (this.isSingle() || !!rootEntityType) || (!!dataReferenceName && !!rootEntityType)
```

Comment:
I believe that it's right. If you're part way through configuring the analysis to run on a snapshot (i.e. you've selected a snapshot name but not a table name) the expression will be `false` because `!!dataReferenceName` = `true`, `!!rootEntityType` = `false`, and `this.isSingle()` will evaluate to `false`

There was a bug that I fixed in a recent commit where the overall expression was evaluating to `true` when it shouldn't have because `rootEntityType` was never unset when selecting a snapshot name, so if you were testing locally that may have been the cause